sway/input/seatop_resize_floating.c

```diff
@@ -55,29 +56,37 @@ static void handle_pointer_motion(struct sway_seat *seat, uint32_t time_msec,
 		grow_height = e->ref_height * max_multiplier;
 	}
 
+	struct sway_container_state *state = &con->current;
+	double border_width = 0.0;
+	if (con->current.border == B_NORMAL || con->current.border == B_PIXEL) {
+		border_width = state->border_thickness * 2;
+	}
+	double border_height = 0.0;
+	if (con->current.border == B_NORMAL) {
+		border_height += container_titlebar_height();
+		border_height += state->border_thickness;
+	} else if (con->current.border == B_PIXEL) {
+		border_height += state->border_thickness * 2;
+	}
+
 	// Determine new width/height, and accommodate for floating min/max values
 	double width = e->ref_width + grow_width;
 	double height = e->ref_height + grow_height;
 	int min_width, max_width, min_height, max_height;
 	floating_calculate_constraints(&min_width, &max_width,
 			&min_height, &max_height);
-	width = fmax(min_width, fmin(width, max_width));
-	height = fmax(min_height, fmin(height, max_height));
+	width = fmax(min_width + border_width, fmin(width, max_width));
```

Comment:
We should probably take `border_width` into account for the max too? Something like `max_width - border_width`? We should also be careful that the result is > 0.

Reply:
Let's merge this PR as-is. I've created a follow-up issue: https://github.com/swaywm/sway/issues/5337